R/ppc-distributions.R

```diff
@@ -270,10 +270,15 @@ ppc_dens <- function(y, yrep, ...,
 
 #' @rdname PPC-distributions
 #' @export
+#' @template args-density-controls
 ppc_dens_overlay <- function(y, yrep, ...,
-                             trim = FALSE,
                              size = 0.25,
-                             alpha = 0.7) {
+                             alpha = 0.7,
+                             trim = FALSE,
+                             bw = "nrd0",
```

Comment:
I'm noticing that the defaults are hard-coded here, but they are deferred in other density functions like `mcmc_areas()`. 

```r
mcmc_areas <- function(..., 
                       bw = NULL,
                       adjust = NULL,
                       kernel = NULL,
                       n_dens = NULL) {}

mcmc_areas_data <- function(...,
                            bw = NULL,
                            adjust = NULL,
                            kernel = NULL,
                            n_dens = NULL) {}

mcmc_areas_ridges <- function(...,
                             bw = NULL, 
                             adjust = NULL, 
                             kernel = NULL,
                             n_dens = NULL){}

mcmc_areas_ridges_data <- function(...,
                                   bw = NULL, 
                                   adjust = NULL, 
                                   kernel = NULL,
                                   n_dens = NULL) {}

# this function does all the density calculations for the four above functions
compute_interval_density <- function(x, interval_width = 1, n_dens = 1024,
                                     bw = NULL, adjust = NULL, kernel = NULL) {
  n_dens <- n_dens %||% 1024
  ...
}

```

We should make this more consistent and update the `@template args-density-controls`. We can make this a separate issue.

Reply:
Agreed!